applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_3D_law.cpp

```diff
@@ -0,0 +1,401 @@
+
+    if (rConstitTensor.size1() != 6 || rConstitTensor.size2() != 6)
+        rConstitTensor.resize(6, 6, false);
+    rConstitTensor.clear();
+
+    rConstitTensor(0, 0) = (E * (1.0 - nu) / ((1.0 + nu) * (1.0 - 2.0 * nu)));
+    rConstitTensor(1, 1) = rConstitTensor(0, 0);
+    rConstitTensor(2, 2) = rConstitTensor(0, 0);
+    rConstitTensor(3, 3) = rConstitTensor(0, 0) * (1.0 - 2.0 * nu) / (2.0 * (1.0 - nu));
+    rConstitTensor(4, 4) = rConstitTensor(3, 3);
+    rConstitTensor(5, 5) = rConstitTensor(3, 3);
+    rConstitTensor(0, 1) = rConstitTensor(0, 0) * nu / (1.0 - nu);
+    rConstitTensor(1, 0) = rConstitTensor(0, 1);
+    rConstitTensor(0, 2) = rConstitTensor(0, 1);
+    rConstitTensor(2, 0) = rConstitTensor(0, 1);
+    rConstitTensor(1, 2) = rConstitTensor(0, 1);
+    rConstitTensor(2, 1) = rConstitTensor(0, 1);
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamage3D::GetLawFeatures(Features& rFeatures)
+{
+    rFeatures.mOptions.Set(THREE_DIMENSIONAL_LAW);
+    rFeatures.mOptions.Set(INFINITESIMAL_STRAINS);
+    rFeatures.mOptions.Set(ISOTROPIC);
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Infinitesimal);
+    rFeatures.mStrainSize = 6;
+    rFeatures.mSpaceDimension = 3;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+int LinearIsotropicDamage3D::Check(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_CHECK(rMaterialProperties.Has(YOUNG_MODULUS));
+    KRATOS_CHECK(rMaterialProperties.Has(POISSON_RATIO));
+    KRATOS_CHECK(rMaterialProperties.Has(YIELD_STRESS));
+    KRATOS_CHECK(rMaterialProperties.Has(INFINITY_YIELD_STRESS));
+    KRATOS_CHECK(rMaterialProperties.Has(ISOTROPIC_HARDENING_MODULUS));
+    KRATOS_CHECK_GREATER(rMaterialProperties[YIELD_STRESS], 0.);
+    KRATOS_CHECK_GREATER(rMaterialProperties[INFINITY_YIELD_STRESS], 0.);
+    KRATOS_CHECK_LESS(rMaterialProperties[ISOTROPIC_HARDENING_MODULUS], 1.);
+    KRATOS_CHECK_NOT_EQUAL(rMaterialProperties[ISOTROPIC_HARDENING_MODULUS], 0.);
+
+    if (rMaterialProperties[ISOTROPIC_HARDENING_MODULUS] > 0 &&
+        rMaterialProperties[INFINITY_YIELD_STRESS] <= rMaterialProperties[YIELD_STRESS])
+        KRATOS_ERROR << "If ISOTROPIC_HARDENING_MODULUS is positive, "
+            "INFINITY_YIELD_STRESS must be greater than YIELD_STRESS" << std::endl;
+
+    if (rMaterialProperties[ISOTROPIC_HARDENING_MODULUS] < 0 &&
+        rMaterialProperties[INFINITY_YIELD_STRESS] >= rMaterialProperties[YIELD_STRESS])
+        KRATOS_ERROR << "If ISOTROPIC_HARDENING_MODULUS is negative, "
+                "INFINITY_YIELD_STRESS must be lesser than YIELD_STRESS" << std::endl;
```

Comment:
smaller

Reply:
lesser doesn't sound good to me either, but I think that lesser and greater are for comparing a relative values, and smaller and bigger for absolute ones.